src/ClientBase.php

```diff
@@ -268,4 +378,52 @@ protected function buildClient(array &$options) {
         unset($options['handler']);
         return $client;
     }
+
+    /**
+     * Helper function to extract the option codes.
+     *
+     * @param array $options
+     *  The options array.
+     *
+     * @return array
+     *  The list of options with the option-code.
+     */
+    protected function parseOptionCodes(array $options)
+    {
+        $valid_options = [];
+        foreach ($options['option_codes'] as $optionCode) {
+            if (!in_array(strtoupper($optionCode), self::getOptionCodes())) {
+                break;
+            }
+            // @todo Perhaps we should check for conflicts here, might be overkill.
+            // From Canada Post docs:
+            // There are some options that can be applied to a shipment that
+            // conflict with the presence of another option. You can use the
+            // "Get Option" call in advance to check the contents of the
+            // <conflicting-options> group from a Get Option call for options
+            // selected by end users or options available for a given service.
+            $valid_options[] = [
+                'option-code' => $optionCode
+            ];
+        }
+
+        return $valid_options;
+    }
+
+    /**
+     * Helper function to verify the postal code for an address.
+     *
+     * Canada Post API requires no spaces and uppercase postal code.
+     *
+     * @param array $address
+     *   The address to verify.
+     */
+    protected function verifyPostalCode($address)
```

Comment:
The way the method is written and called implies it expects the address to be passed in reference, but it is not. Is it meant to be `&$address`?